### PR TITLE
the same old prettyPrint() rework, but now without merging conflicts

### PR DESCRIPTION
**Pull request by TerminalMan**
I "rebased" my changes onto your develop branch head. I hope it is bug free now. The cards i tested it with worked fine.

I noticed you reworked the UI. I have to say i prefer the old "delete and overwrite" UI with colors and less enter smashing. Reasons why i preferred the previous version:

- less terminal clutter
- you can't accidentally look onto cards you just did to get the answer
- It's now more difficult to focus on the new card, you have to look for it, before the new Card always started exactly at the same location

Aside from the terminal corruption bug, when running the program via 'go run nao.go' instead of 'go build nao.go && ./go' it felt solid.

### CODE REVIEW

**Comment by gRastello** [comment]
Yeah I'm still skeptic about the "new" UI myself. I definitely want to avoid the hackish call to `stty` I was using before so smashing enter is probably going to be a thing but I'm planning to implement some way to clear the screen after a flashcard has been reviewed.

It wasn't necessary to rebase your changes but you saved me the work to solve a merge conflict. I definitely appreciate it. I'll look and test your changes in the following days and I'll let you know if everything seems fine to me.

**Comment by gRastello** [comment]
So first of all you should use import `github.com/mattn/go-runewidth` rather than just `go-runewidth` since the former will search for the package in `/usr/lib/go/src/go-runewidth` and `$GOPATH/src/go-runewidth` while `go-runewidth` is actually installed (by conventional means) in `$GOPATH/src/github.com/mattn/go-runewidth`.  But that's a minor thing.

I've done some testing and found out that line breaks with `<br>` works as intended, long lines are split correctly even when they contain huge amounts of unicode. I'll leave you a couple of comments to review.

File: nao.go [hunk 1]
```diff
@@ -6,11 +6,11 @@ import (
 	"math"
 	"math/rand"
 	"os"
-	"os/exec"
 	"os/user"
 	"strconv"
 	"strings"
 	"time"
+	"go-runewidth"
```
Comment (gRastello) [inline nao.go]: I've talked about this in the previous post.

File: nao.go [hunk 2]
```diff
@@ -123,34 +124,31 @@ func getKey(prompt string) byte {
 }
 
 // print function designed for handling long flashcards in a nice way
-func prettyPrint(s1, s2 string) {
-	s1n := len(s1)
+func prettyPrint(s1, s2 string) int {
```
Comment (gRastello) [inline nao.go]: As you can see by inspecting the current state of the `develop` branch `prettyPrint` doesn't return an integer anymore so just remove this `int`, the return statement and the `lines` variable from the function.

File: nao.go [hunk 3]
```diff
@@ -123,34 +124,31 @@ func getKey(prompt string) byte {
...
-			s2 = s2[LINELENGTH-s1n:]
-			fmt.Printf("%s\n", ss)
+		if col > LINELENGTH {
+			fmt.Printf("\n")
+			fmt.Printf("       ")
+			col = len(s1)+1
+			lines++
 		}
 	}
+	fmt.Printf("\n")
```
Comment (gRastello) [inline nao.go]: This newline mess with the "new" UI that I'm testing in the `develop` branch. Nothing too serious but this line in particular adds unnecessary line breaks to Nao's output.

File: nao.go [hunk 4]
```diff
@@ -123,34 +124,31 @@ func getKey(prompt string) byte {
...
-
-	for len(s2) > 0 {
-		for i := 0; i < s1n; i++ {
-			fmt.Printf(" ")
+	col := len(s1)+1
+	for _, rune := range s2 {
+		fmt.Printf("%c", rune)
+		col += runewidth.RuneWidth(rune)
+		if rune == '\n' {
+			fmt.Printf("       ")
```
Comment (gRastello) [inline nao.go]: I understand why you did this but please rollback to the old for loop or a similar alternative that takes into consideration the length of `s1`. `prettyPrint` was designed to be a general way to print two strings in a nice way and not a particular way of printing two strings (in this case "Front: " or "Back: " and another string). `prettyPrint` could in theory be used in other context (and can definitely be the case in the future) so it's better if it remains as generic as possible.

File: nao.go (hunk 1 above)
Comment (TerminalMan) [reply nao.go]: the reason I went with "go-runewidth" was when i 'git clone' the go-runewidth archive into ~/.go/src/ it ends up becoming '~/.go/src/go-runewidth'. And next to the global /usr/ path this seems to be the standard local user go library path. I may be wrong with this, I am not familiar at all with go.

The reason for removing the s1n was that the initial version i had didn't require it at all and i kind of build the then committed version on top of that one. I guess the benefit is that the code is one line shorter :p

File: nao.go (hunk 2 above)
Comment (TerminalMan) [reply nao.go]: Seeing that you aren't totally convinced about the UI rework, wouldn't it make sense to keep the 'return line_count', thus easily allowing to return to the 'overwrite the old text route'?

Comment (gRastello) [reply nao.go]: Makes sense, let's keep this then

File: nao.go (hunk 1 above)
Comment (gRastello) [reply nao.go]: Well as I've said above I prefer having an extra line of code to calling a function multiple times to get the value every time.

File: nao.go (hunk 4 above)
Comment (TerminalMan) [reply nao.go]: I take it your issue is the static "fmt.Printf("       ")" and that the spacing should be a function of len(s1n).

File: nao.go (hunk 1 above)
Comment (TerminalMan) [reply nao.go]: Yes, I got that. What I meant to say was, I'm not disagreeing with your logic. I merely tried to answer the 'why did you delete...' in your question.

File: nao.go (hunk 4 above)
Comment (gRastello) [reply nao.go]: That is it man!

**Comment by gRastello** [comment]
Changes approved and merged to `develop`. As a last thing: you forgot to run your code through `gofmt` but that's no big deal; I've run the command for you ;).

**Comment by TerminalMan** [comment]
Sorry, I didn't even know that was a thing :X

**Pull request closed without merging**